csrc/multidevice/utils.cpp

```diff
@@ -177,6 +177,21 @@ bool isResharding(const Expr* expr) {
     return false;
   }
 
+  // Reduction over a sharded dimension.
+  if (expr->isA<ReductionOp>()) {
+    // auto in = expr->as<ReductionOp>()->in()->as<TensorView>();
+    auto out = expr->as<ReductionOp>()->out()->as<TensorView>();
+    std::vector<IterDomain*> reduction_axis;
+    std::copy_if(
+        out->getLogicalDomain().begin(),
+        out->getLogicalDomain().end(),
+        std::back_inserter(reduction_axis),
+        [](IterDomain* id) { return id->isReduction(); });
+    if (reduction_axis[0]->isDeviceDim()) {
+      return true;
+    }
```

Comment:
```suggestion
    if (std::any_of(out_logical.begin(), out_logical.end(), std::mem_fn(IterDomain::isReduction)) {
      return true;
    }
```
It's not doing exactly the same thing but I think it's simpler and more general -- it seems unnecessary to only check whether the first reduction axis is DID.